### ISSUE TITLE
iree-compile crash on decoder StableHLO export: const-fold null-deref (frozen K activations)

### ISSUE THREAD

**Issue by michalharakal** [sk-ai-net/skainet#759]
## Summary
Exporting a real decoder (TinyLlama, seqLen>1) to StableHLO and running `iree-compile` crashes with a **null-deref in `mlir::ElementsAttr::getType()` during constant folding** (`ConvertTensorToFlowPass`, input→flow). Reproduces on IREE **3.7.0 / 3.10.0 / 3.11.0** (version-independent).

## Backtrace (with `--mlir-disable-threading`)
```
mlir::ElementsAttr::getType() const + 4          <-- SIGSEGV
... mlir::Operation::fold(...)
... mlir::applyPatternsGreedily(...)
ConvertTensorToFlowPass (iree-dispatch-creation-convert-tensor-to-flow)
```

## Minimal repro (the exact crashing region, dumped before the pass)
```mlir
util.initializer {
  %0 = flow.tensor.constant #flow.parameter.named<"model"::"t27"> : tensor<4x8x64xf32>
  %extracted_slice = tensor.extract_slice %0[0,0,0][1,8,64][1,1,1] : tensor<4x8x64xf32> to tensor<8x64xf32>
  %1 = tensor.empty() : tensor<32x8x64xf32>
  %inserted_slice = tensor.insert_slice %extracted_slice into %1[0,0,0][1,8,64][1,1,1] : tensor<8x64xf32> into tensor<32x8x64xf32>
  util.global.store %inserted_slice, @__hoisted_tensor_32x8x64xf32 : tensor<32x8x64xf32>
  util.return
}
```
IREE folds `insert_slice(extract_slice(constant))` and reads a null `ElementsAttr`.

## Two root causes (one in IREE, one in our export)
1. **IREE bug:** the `insert_slice`/`extract_slice` fold over a `flow.tensor.constant` null-derefs. Worth reporting upstream (the IR above is valid).
2. **Our export freezes activations as constants (the real fix on our side).** `t27` is `tensor<4x8x64xf32>` = `[nKVHeads, seq, headDim]` — a **K activation**, not a weight, yet it's materialized as an external parameter. Census of the exported params: **44 of 201 carry the `seq` dim** (`[4,8,64]`×22 + `[32,8,64]`×22) — the K / GQA-`repeatKVHeads` path is being **const-folded** under `VoidTensorOps` + `embedConstants=true`. Those frozen constants then feed the `repeatKVHeads` slice/insert expansion that IREE chokes on. Q and V are not frozen — only the K path (RoPE + repeatKVHeads) is, which is the lead to chase.

## What's already fixed (PR linked)
- **Graph-output pruning** (`ComputeGraph.prunedToOutputs`): the export was returning the logits + ~44 dangling per-layer outputs; now pruned to logits only (45→1 outputs, 1467→1247 nodes, attention intact). Removes the dangling-output contribution but **not** the crash above.
- **SDPA causal mask** emits `-1e30` instead of `-inf` (hygiene).

## Remaining work
Stop the K/RoPE/`repeatKVHeads` path from being const-folded into parameters during the `VoidTensorOps` + `embedConstants` export (so K stays a runtime activation; its `repeatKVHeads` slice/insert then operates on a value, not a hoisted constant). Likely the same family as the Llama `fromWeights`/DSL tracing-fidelity issue.

## Repro steps
```
# export (skainet-tinyllama-iree harness) then:
iree-compile tinyllama_iree.mlir --compile-to=flow -o /dev/null --mlir-disable-threading   # crashes
```
